fms/utils/serialization.py

```diff
@@ -484,10 +491,20 @@ def _load_partial_state_dict(
             # into the model
             if not needs_tp_sharding or tp_module is None:
                 param = getattr(target_module, key_steps[-1])
+                if param.device == torch.device("meta"):  # TODO: improve this clunky cast from meta to cpu/gpu
+                    if isinstance(param, torch.nn.Parameter):
+                        param = torch.nn.Parameter(torch.empty_like(param, device=tensor_value.device))
+                    else:
+                        param = torch.empty_like(param, device=tensor_value.device)
+                    setattr(target_module, key_steps[-1], param)
+                    param = getattr(target_module, key_steps[-1])
                 param.copy_(tensor_value, non_blocking=True)
             elif tp_module is not None and tp_module not in seen_tp_modules:
                 seen_tp_modules.add(tp_module)
                 tensor_values = {k: v for k, v in state_dict.items() if tp_prefix in k}
-                tp_module.load_weights(tensor_values)
+                if "gptq_hf" in source:
+                    tp_module.load_qparams(tensor_values)
+                else:
+                    tp_module.load_weights(tensor_values)
```

Comment:
pass model.config to load_weights to discover type of quantization, default to regular linear layers if config is non-existent